### PR TITLE
Create 0101-symmetric-tree.js

### PR DESCRIPTION
**Pull request by aadil42**
Solved symmetric-tree in JS.

File(s) Added: 0101-symmetric-tree.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/symmetric-tree/submissions/1346787470/

### CODE REVIEW

File: javascript/0101-symmetric-tree.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * DFS | Tree-trevarsal
+ * Time O(n) | Space O(n)
+ * https://leetcode.com/problems/symmetric-tree/
+ * @param {TreeNode} root
+ * @return {boolean}
+ */
+var isSymmetric = function(root) {
+    
+    const dfs = (node1, node2) => {
```
Comment (aakhtar3) [inline javascript/0101-symmetric-tree.js:18]: Avoid nested functions

```js

/**
 * DFS - Post Order
 * Time O(N) | Space O(N)
 * https://leetcode.com/problems/symmetric-tree/
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = (root) => isMirror(root, root);

const isMirror = (left, right) => {
    const isBaseCase1 = !(left && right);
    if (isBaseCase1) return (left === right);
    
    const isBaseCase2 = !(left.val === right.val);
    if (isBaseCase2) return false;
    
    const leftMirror = isMirror(left.left, right.right)
    const rightMirror = isMirror(left.right, right.left)
    
    return leftMirror && rightMirror
}
```

Comment (aadil42) [reply javascript/0101-symmetric-tree.js:18]: Omitted nested function.
Submission link for the proposed code: https://leetcode.com/problems/symmetric-tree/submissions/1388706138/

```javascript
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * DFS | Tree-trevarsal
 * Time O(n) | Space O(n)
 * https://leetcode.com/problems/symmetric-tree/
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    return dfs(root.left, root.right);
};
    
const dfs = (node1, node2) => {

    if(!node1 && !node2) return true;

    if(node1 && !node2) return false;
    if(!node1 && node2) return false;
    if(node1.val !== node2.val) return false;

    return dfs(node1.right, node2.left) && dfs(node1.left, node2.right);
}
```

Comment (aadil42) [reply javascript/0101-symmetric-tree.js:18]: @aakhtar3 (tagging to draw attention to the proposed code)

Comment (aadil42) [reply javascript/0101-symmetric-tree.js]: @Ykhan799, I fixed the spacing and updated the code.

**Pull request merged**